src/vellum/workflows/errors/types.py

```diff
@@ -15,6 +21,42 @@ class VellumErrorCode(Enum):
 
 
 @dataclass(frozen=True)
-class VellumError:
+class WorkflowError:
     message: str
-    code: VellumErrorCode
+    code: WorkflowErrorCode
+
+
+_VELLUM_ERROR_CODE_TO_WORKFLOW_ERROR_CODE: Dict[VellumErrorCodeEnum, WorkflowErrorCode] = {
+    "INVALID_REQUEST": WorkflowErrorCode.INVALID_INPUTS,
+    "INVALID_INPUTS": WorkflowErrorCode.INVALID_INPUTS,
+    "PROVIDER_ERROR": WorkflowErrorCode.PROVIDER_ERROR,
+    "REQUEST_TIMEOUT": WorkflowErrorCode.PROVIDER_ERROR,
+    "INTERNAL_SERVER_ERROR": WorkflowErrorCode.INTERNAL_ERROR,
+    "USER_DEFINED_ERROR": WorkflowErrorCode.USER_DEFINED_ERROR,
+}
+
+
+def vellum_error_to_workflow_error(error: VellumError) -> WorkflowError:
+    return WorkflowError(
+        message=error.message,
+        code=_VELLUM_ERROR_CODE_TO_WORKFLOW_ERROR_CODE.get(error.code, WorkflowErrorCode.INTERNAL_ERROR),
+    )
+
+
+_WORKFLOW_EVENT_ERROR_CODE_TO_WORKFLOW_ERROR_CODE: Dict[WorkflowExecutionEventErrorCode, WorkflowErrorCode] = {
+    "WORKFLOW_INITIALIZATION": WorkflowErrorCode.INVALID_WORKFLOW,
+    "WORKFLOW_CANCELLED": WorkflowErrorCode.WORKFLOW_CANCELLED,
+    "NODE_EXECUTION_COUNT_LIMIT_REACHED": WorkflowErrorCode.INVALID_STATE,
+    "INTERNAL_SERVER_ERROR": WorkflowErrorCode.INTERNAL_ERROR,
+    "NODE_EXECUTION": WorkflowErrorCode.PROVIDER_ERROR,
```

Comment:
This mapping here surprises me. Are all node execution errors due to provider errors? Seems unlikely

Reply:
They're not, but was having trouble which to map it to, or to introduce a new one